docs/src/main/asciidoc/spring-cloud-stream.adoc

```diff
@@ -709,7 +709,60 @@ public Function<List<Person>, Person> findFirstPerson() {
 }
 ----
 
-==== Annotation-based support
+===== Spring Integration flow as functions
+There are times when implementing a function you may have complex requirements that fit the category of
+https://www.enterpriseintegrationpatterns.com[Enterprise Integration Patterns] (EIP) and as such
+best to be handled using framework such as https://spring.io/projects/spring-integration[Spring Integration] (SI) -
+a reference implementation of EIP.
+
+Thankfully SI already provides support for exposing integration flows as functions.
+Let's look at the sample:
+
+[source, java]
+----
+@SpringBootApplication
+public class FunctionSampleSpringIntegrationApplication {
+
+	public static void main(String[] args) {
+		SpringApplication.run(FunctionSampleSpringIntegrationApplication.class, args);
+	}
+
+	@Bean
+	public IntegrationFlow uppercaseFlow() {
+		return IntegrationFlows.from(MessageFunction.class, "uppercase")
+				.<String, String>transform(String::toUpperCase)
+				.logAndReply(LoggingHandler.Level.WARN);
+	}
+
+	public interface MessageFunction extends Function<Message<String>, Message<String>> {
+
+	}
+}
+----
+
+For those who are familiar with SI you can see we are simply defining a bean of type `IntegrationFlow` where we
+declaring an integration flow we want to expose as a `Function<String, String>` (using SI DSL) by the name `uppercase`.
+The `MessageFunction` interface here simply allows us to explicitly declare the type of the inputs and outputs for
+the purpose of proper type conversion.
+For cases where you simply want to receive raw input you could use `from(Function.class, ...)`.
+
+The resulting function will be bound to the input and output destination exposed by the target binder.
+For simple cases you don't need to now the names of such destinations, however for most cases where you want to configure
+destination's Producer/Consumer properties you do need to know such names. For cases where this is the only function
+that you intend to use for bindings, the names are `input` and `output`. For cases where you may have multiple bindings (<<Multiple functions in a single application>>)
+the binding naming conventions is as follows:
+
+*Binding naming convention:*
+
+* input - `<functionName> + .in. + <index>`
+* output - `<functionName> + .out. + <index>`
+
+So for single binding cases an example of configuration property for input is `spring.cloud.stream.bindings.input.destination=myQueue` and for
```

Comment:
Add a comma after "so", hyphenate "single-binding", and add a comma after "cases".